### PR TITLE
newrelic-cli 0.2.3 (new formula)

### PR DESCRIPTION
**Pull request by zlesnr**
New Relic is in the process of developing a CLI to interact with New Relic SaaS
infrastructure.  Here we include a new formula that will download and install
the pre-release version.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by zlesnr** [comment]
Dearest maintainers, might we have some forgiveness here for a new project?  We expect to be increasing our notability here in the coming weeks, but having a brew ahead of time would perhaps help our adoption.

File: Formula/newrelic.rb
Comment (Bo98) [inline Formula/newrelic.rb]: As per https://docs.brew.sh/Acceptable-Formulae#we-dont-like-binary-formulae, this needs to be built from source.

**Review by Bo98 — CHANGES_REQUESTED** [review]

File: Formula/newrelic.rb
Comment (zlesnr) [reply Formula/newrelic.rb]: Thank you for the feedback.  I've updated the port to build from source.

File: Formula/newrelic.rb
Comment (zlesnr) [reply Formula/newrelic.rb]: Just updated.

File: Formula/newrelic.rb
Comment (zlesnr) [reply Formula/newrelic.rb]: Updated.

**Review by chenrui333 — APPROVED** [review]
❤️

File: Formula/newrelic.rb
Comment (jthurman42) [inline Formula/newrelic.rb]: We should rename this to `NewrelicCli` and change the file to `newrelic-cli` as well.  `newrelic` is probably too general at this time.

File: Formula/newrelic.rb
Comment (jthurman42) [inline Formula/newrelic.rb]: `version` is a method, and testing shows that leaving the version in the URL it does properly detect.  So instead of setting it, we should revert back to the auto-detection and the lower L14 change still works as expected.

**Review by jthurman42 — CHANGES_REQUESTED** [review]

File: Formula/newrelic.rb
Comment (zlesnr) [reply Formula/newrelic.rb]: Done.

**Comment by zlesnr** [comment]
We've made a few adjustments here, but this should be good to go from our side.  May I have another review please @Bo98?

File: Formula/newrelic-cli.rb
Comment (Bo98) [inline Formula/newrelic-cli.rb]: Do you support Go modules? If so, you could simplify a lot of this (example formula which moved to use Go modules: https://github.com/Homebrew/homebrew-core/commit/8b7ebcbbcbf5fa60677e1ae8d7b59324bd83a2ce).

File: Formula/newrelic-cli.rb
Comment (Bo98) [inline Formula/newrelic-cli.rb]: It's better than nothing, but `--help` tests really should be avoided. See [here](https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula) for general guidance, including suggestions for cases where testing may be trickier.

**Review by Bo98 — COMMENTED** [review]
I'll be doing a pass over new formulae over the weekend. I also want to make sure other maintainers have had a chance to comment since your repository is currently under the notability line.

Nevertheless, here's some feedback on the formula file.

File: Formula/newrelic-cli.rb
Comment (zlesnr) [reply Formula/newrelic-cli.rb]: We do support go modules, I wasn't aware that we could remove all of that.  Cleaner now.

File: Formula/newrelic-cli.rb
Comment (zlesnr) [reply Formula/newrelic-cli.rb]: I've replaced this with a `newrelic config list` command.  Most of the commands here will require an API key to really validate.  But if we can list the users config (present or not), then that is at least exercising something.

File: Formula/newrelic-cli.rb
Comment (SMillerDev) [inline Formula/newrelic-cli.rb]: Pleas check some output here using `assert_match` and `shell_output`

File: Formula/newrelic-cli.rb
Comment (SMillerDev) [inline Formula/newrelic-cli.rb]: Does this download any dependencies?

Comment (zlesnr) [reply Formula/newrelic-cli.rb]: Yes, the go modules.

File: Formula/newrelic-cli.rb
Comment (zlesnr) [reply Formula/newrelic-cli.rb]: I've done as you ask to include a version check of the output.

**Comment by zlesnr** [comment]
Thank you for the review.  Let me know if anything else needs adjusting @SMillerDev.

File: Formula/newrelic-cli.rb
Comment (zlesnr) [reply Formula/newrelic-cli.rb]: This has since been moved to `version` so that the output can be checked reliably.

File: Formula/newrelic-cli.rb
Comment (SMillerDev) [inline Formula/newrelic-cli.rb]: We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
Same if the software requires a virtual machine, docker instance, etc. to be running.

Comment (zlesnr) [reply Formula/newrelic-cli.rb]: Thanks for the feedback.  I've included a few other checks for a command that is not help and not version, as requested.  The trouble we have is that most of this tool requires an API key to be functional.  Anyway, I appreciate the review.  Let me know if that is sufficient.

**Comment by SMillerDev** [comment]
Thanks @zlsner! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by Bo98** [comment]
Before merging, just some style complaints:

```
==> brew audit newrelic-cli --online --new-formula
==> FAILED
newrelic-cli:
  * C: 2: col 8: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 3: col 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 4: col 7: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 5: col 10: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 6: col 8: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 8: col 14: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 11: col 9: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 12: col 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 12: col 20: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
  * C: 13: col 17: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
Error: 10 problems in 1 formula detected
```

Use double quotes over single quotes.

**Comment by zlesnr** [comment]
I'll fix those.  Each update I made, the quotes get set by my editor to follow a different standard.

**Comment by zlesnr** [comment]
Just updated @Bo98.

**Review by Bo98 — APPROVED** [review]

**Pull request closed without merging**
